### PR TITLE
[infra] Support syslog rate limit configuration

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
HLD: https://github.com/sonic-net/SONiC/pull/1049

Depends on https://github.com/sonic-net/sonic-buildimage/pull/12489 (Build dependency)

#### Why I did it

Support syslog rate limit configuration feature

#### How I did it

1. Remove unused rsyslog.conf from containers
2. Modify docker startup script to generate rsyslog.conf from template files
3. Add metadata/init data for syslog rate limit configuration

#### How to verify it

Manual test
New sonic-mgmt regression cases

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

**Review by keboliu — APPROVED** [review]

File: files/build_templates/init_cfg.json.j2 [hunk 1]
```diff
@@ -99,6 +100,20 @@
             "rate_limit_interval" : "600",
             "available_mem_threshold": "10.0"
         }{%if not loop.last %},{% endif -%}
+{% endfor %}
+    },
+    "SYSLOG_CONFIG": {
+        "GLOBAL": {
+            "rate_limit_interval" : "0",
```
Comment (mkeda) [inline files/build_templates/init_cfg.json.j2:107]: Just to confirm - you do not want rate limiting on host?

File: files/image_config/rsyslog/rsyslog-container.conf.j2 [hunk 2]
```diff
@@ -14,8 +14,24 @@ $ModLoad imuxsock # provides support for local system logging
...
+{% endif %}
+{% endif %}
+{% endif %}
+
+{% if rate_limit_interval is defined %}
+$SystemLogRateLimitInterval {{ rate_limit_interval }}
+{% endif %}
+{% if rate_limit_burst is defined %}
+$SystemLogRateLimitBurst {{ rate_limit_burst }}
+{% endif %}
```
Comment (mkeda) [inline files/image_config/rsyslog/rsyslog-container.conf.j2:34]: Do you want an else case to provide default values if the feature is not on?

File: files/image_config/rsyslog/rsyslog.conf.j2 [hunk 3]
```diff
@@ -14,6 +14,25 @@
...
+{% endif %}
+{% endif %}
+{% endif %}
+
+{% if rate_limit_interval is defined %}
+$SystemLogRateLimitInterval {{ rate_limit_interval }}
+{% endif %}
+{% if rate_limit_burst is defined %}
+$SystemLogRateLimitBurst {{ rate_limit_burst }}
+{% endif %}
```
Comment (mkeda) [inline files/image_config/rsyslog/rsyslog.conf.j2:34]: Same question about else case for providing default values if the feature is not defined.

**Review by mkeda — APPROVED** [review]
Thank you for your work to add this feature!

File: files/build_templates/init_cfg.json.j2 (hunk 1 above)
Comment (Junchao-Mellanox) [reply files/build_templates/init_cfg.json.j2:107]: Yes, host side has no rate limiting by default.

File: files/image_config/rsyslog/rsyslog.conf.j2 (hunk 3 above)
Comment (Junchao-Mellanox) [reply files/image_config/rsyslog/rsyslog.conf.j2:34]: If feature is not defined, there is no configuration of `$SystemLogRateLimitBurst`, `$SystemLogRateLimitInterval`. In that case, there is no rate limiting.

File: files/image_config/rsyslog/rsyslog-container.conf.j2 (hunk 2 above)
Comment (Junchao-Mellanox) [reply files/image_config/rsyslog/rsyslog-container.conf.j2:34]: If feature is not defined, there is no configuration of `$SystemLogRateLimitBurst`, `$SystemLogRateLimitInterval`. In that case, there is no rate limiting.

**Comment by Junchao-Mellanox** [comment]
It has a build dependency on https://github.com/sonic-net/sonic-buildimage/pull/12489 , once that PR is merged, I will rebase this one.

**Review by saiarcot895 — APPROVED** [review]

**Review by saiarcot895 — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]

**Comment by Junchao-Mellanox** [comment]
Created backport PR https://github.com/sonic-net/sonic-buildimage/pull/13535

**Pull request merged**